lib/cocoapods/target/pod_target.rb

```diff
@@ -712,7 +712,7 @@ def recursive_dependent_targets(configuration: nil)
         hash[nil] = hash.each_value.reduce(Set.new, &:|).to_a
         hash
       end
-      @recursive_dependent_targets[configuration]
+      @recursive_dependent_targets.fetch(configuration) { raise ArgumentError, "No configuration #{configuration} for #{self}, known configurations are #{@recursive_dependent_targets.keys}" }
```

Comment:
does the same fix need to be applied to `recursive_test_dependent_targets` below?

Reply:
nope, because that's only called by methods that also call `recursive_dependent_targets`